apps/shipments/admin.py

```diff
@@ -123,4 +135,156 @@ def has_change_permission(self, request, obj=None):
         return False
 
 
-admin.site.register(Shipment, ShipmentAdmin)
+class BaseModelHistory(SimpleHistoryAdmin):
+
+    def history_view(self, request, object_id, extra_context=None):     # pylint:disable=too-many-locals
+        request.current_app = self.admin_site.name
+        model = self.model
+        opts = model._meta
+        app_label = opts.app_label
+        pk_name = opts.pk.attname
+        history = getattr(model, model._meta.simple_history_manager_attribute)
+        object_id = unquote(object_id)
+        action_list = history.filter(**{pk_name: object_id})
+
+        history_list_display = getattr(self, "history_list_display", [])
+        # If no history was found, see whether this object even exists.
+        try:
+            obj = self.get_queryset(request).get(**{pk_name: object_id})
+        except model.DoesNotExist:
+            try:
+                obj = action_list.latest("history_date").instance
+            except action_list.model.DoesNotExist:
+                raise http.Http404
+
+        if not self.has_change_permission(request, obj):
+            # We override this behavior for the moment
+            # raise PermissionDenied
+            pass
+
+        for history_list_entry in history_list_display:
+            value_for_entry = getattr(self, history_list_entry, None)
+            if value_for_entry and callable(value_for_entry):
+                for list_entry in action_list:
+                    setattr(list_entry, history_list_entry, value_for_entry(list_entry))
+
+        content_type = ContentType.objects.get_by_natural_key(*USER_NATURAL_KEY)
+        admin_user_view = "admin:%s_%s_change" % (
+            content_type.app_label,
+            content_type.model,
+        )
+        context = {
+            "title": _("Change history: %s") % force_text(obj),
+            "action_list": action_list,
+            "module_name": capfirst(force_text(opts.verbose_name_plural)),
+            "object": obj,
+            "root_path": getattr(self.admin_site, "root_path", None),
+            "app_label": app_label,
+            "opts": opts,
+            "admin_user_view": admin_user_view,
+            "history_list_display": history_list_display,
+        }
+        context.update(self.admin_site.each_context(request))
+        context.update(extra_context or {})
+        extra_kwargs = {}
+        return render(request, self.object_history_template, context, **extra_kwargs)
+
+    def history_form_view(self, request, object_id, version_id, extra_context=None):    # pylint:disable=too-many-locals
```

Comment:
Please make every effort to follow pylint's rules instead of disabling the checks